x-pack/plugins/index_management/public/application/components/mappings_editor/components/document_fields/field_parameters/select_inference_id.tsx

```diff
@@ -112,7 +112,14 @@ export const SelectInferenceId = ({
   }, [models]);
 
   useEffect(() => {
-    setOptions([...defaultInferenceIds, ...inferenceIdOptionsFromModels]);
+    const mergedOptions = {
```

Comment:
Can we add removing duplicates logic when constructing [Default inference IDs](https://github.com/elastic/kibana/blob/main/x-pack/plugins/index_management/public/application/components/mappings_editor/components/document_fields/field_parameters/select_inference_id.tsx#L98) ? 

we cannot create duplicate inference id via flyout or via API. The only possibility to show duplicate values is when default inference id such as elser_model_2 and e5 is not downloaded & deployed. But is added via API. WDYT?

Reply:
@saarikabhasi In this case, duplicates are occurring even when both the elser_model_2 and e5 trained models are downloaded and deployed. The root cause of these duplicates is the presence of inference IDs named 'elser_model_2' or 'e5'. The current logic checks whether the user has already created an inference endpoint named 'elser_model_2' or 'e5'. If so, the logic merges the other attributes of default_inference_ids.